src/events.ts

```diff
@@ -65,19 +80,24 @@ export abstract class EventEmitter {
 
     /**
      * Gets an array of listeners for a specific event type.
-     * @param eventName {string} eventName The type of the event.
+     * @param {string} eventName eventName The type of the event.
      */
     public listeners(eventName: string): ((event: EventArgs) => void)[] {
         return (this.eventListeners[eventName] || []);
     }
 
     /**
      * Invokes each listener registered for the specified event type.
-     * @param eventName {string} eventName The type of the event.
+     * @param {string} eventName The type of the event.
+     * @param {MessageBus} ipc (Optional) The {MessageBus} in which to broadcast the event.
      */
-    public emit(eventName: string, eventArgs: EventArgs) {
-        for (const listener of this.listeners(eventName)) {
-            listener(eventArgs);
+    public emit(eventName: string, eventArgs: EventArgs, ipc?: MessageBus) {
```

Comment:
Add optional ipc param.  If present, forward the event via the bus instead of directly emitting.  If on the bus, each instance of the eventemitter created around the bus will receive the message and translate it to an emit.

Reply:
This info probably should be added to the docs

Reply:
It is, look at the full diff